File: mods/drift-legacy/src/main/java/gg/drift/legacy/mixin/InGameHudMixin.java
package gg.drift.legacy.mixin;

import gg.drift.legacy.hud.*;
import net.minecraft.client.gui.hud.InGameHud;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(InGameHud.class)
public class InGameHudMixin {

    @Inject(method = "render", at = @At("TAIL"))
    public void onRender(float tickDelta, CallbackInfo ci) {
        LegacyFpsCounter.render();
        LegacyPingDisplay.render();
        LegacyCpsCounter.render();
        LegacyKeystrokes.render();
    }
}

File: mods/drift-qol/src/main/kotlin/gg/drift/qol/mixin/InGameHudMixin.kt
package gg.drift.qol.mixin

import gg.drift.qol.features.CustomCrosshair
import net.minecraft.client.gui.hud.InGameHud
import net.minecraft.client.gui.DrawContext
import net.minecraft.client.render.RenderTickCounter
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo

@Mixin(InGameHud::class)
class InGameHudMixin {

    @Inject(method = ["render"], at = [At("TAIL")])
    fun onRender(context: DrawContext, tickCounter: RenderTickCounter, ci: CallbackInfo) {
        // Crosshair rendering is handled by HudRenderCallback in QoLKeybinds
        // This mixin is reserved for future QoL HUD overlays
    }

    @Inject(method = ["renderCrosshair"], at = [At("HEAD")], cancellable = true)
    fun onRenderCrosshair(context: DrawContext, tickCounter: RenderTickCounter, ci: CallbackInfo) {
        if (CustomCrosshair.enabled) {
            ci.cancel() // Suppress vanilla crosshair when custom is active
        }
    }
}
